### PR TITLE
Remove StorageExtensions

### PR DESCRIPTION
**Pull request by twz123**
## Description

It's deprecated and marked to be ignored in 1.30.

See:
* #3651

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: cmd/config/validate_test.go [hunk 1]
```diff
@@ -35,9 +35,9 @@ func validConfig() []byte {
 
 func invalidConfig() []byte {
 	config := v1beta1.DefaultClusterConfig()
-	config.Spec.Extensions.Storage.Type = "external_storage"
-	//  CreateDefaultStorageClass is not supported with external_storage
-	config.Spec.Extensions.Storage.CreateDefaultStorageClass = true
+	// NLLB cannot be used with external address
+	config.Spec.Network.NodeLocalLoadBalancing.Enabled = true
+	config.Spec.API.ExternalAddress = "k0s.example.com"
```
Comment (jnummelin) [inline cmd/config/validate_test.go:40]: hmm, is this really relevant for this PR? If removing openebs results this test being invalid/not-needed, maybe we should just remove the test?

File: pkg/apis/k0s/v1beta1/extensions.go [hunk 2]
```diff
@@ -28,9 +28,22 @@ var _ Validateable = (*ClusterExtensions)(nil)
 
 // ClusterExtensions specifies cluster extensions
 type ClusterExtensions struct {
-	//+kubebuilder:deprecatedversion:warning="storage is deprecated and will be ignored in 1.30. https://docs.k0sproject.io/stable/examples/openebs".
-	Storage *StorageExtension `json:"storage"`
-	Helm    *HelmExtensions   `json:"helm"`
+	// Deprecated: storage is deprecated and will be ignored starting from k0s
+	// 1.31 and onwards: https://docs.k0sproject.io/stable/examples/openebs
```
Comment (jnummelin) [inline pkg/apis/k0s/v1beta1/extensions.go:32]: In 1.30 we already say it's being ignored, now we're saying we'll ignore it 1.31 onwards. bit confusing

File: cmd/config/validate_test.go (hunk 1 above)
Comment (twz123) [reply cmd/config/validate_test.go:40]: The test checks the config validation in general, it's not specific to OpenEBS, but used it as an example for an invalid config. So I just picked the first other invalid config combination that came to my mind.

File: pkg/apis/k0s/v1beta1/extensions.go (hunk 2 above)
Comment (twz123) [reply pkg/apis/k0s/v1beta1/extensions.go:32]: We said "and _will be_ ignored in 1.30" (emphasis mine), but that never happened. This PR is now fulfilling that promise.

**Review by jnummelin — APPROVED** [review]

**Pull request merged**